### PR TITLE
Fix NO_EXIT_RUNTIME + USE_PTHREADS + MODULARIZE

### PR DESCRIPTION
**Pull request by kripken**
When NO_EXIT_RUNTIME the local noExitRuntime must
be set. In pthreads mode, the workers load the main JS
using `importScripts` which puts the main JS's noExitRuntime
into worker.js's global scope, so the local has the right
value. However, in modularize mode the imported script
has its contents inside a function scope. So we were not
seeing the proper value of noExitRuntime.

This makes us call a helper function, exported from the
module, to access the value of noExitRuntime.

@juj I see that the meaning of noExitRuntime is different
for pthreads, and I'm not entirely sure why - the postamble.js
code says
```
if (!ENVIRONMENT_IS_PTHREAD) // EXIT_RUNTIME=0 only applies to default behavior of the main browser thread
  noExitRuntime = true;
```
As a result pthreads have noExitRuntime set to false
even if the user built with NO_EXIT_RUNTIME.
To keep them alive, one must call something that sets
it to true manually (like exit_with_live_runtime, or
an async function like wget). Is that intentional?

### CODE REVIEW

**Comment by VirtualTim** [comment]
I wonder if this will fix #8202?

**Comment by juj** [comment]
https://github.com/emscripten-core/emscripten/pull/10126 fixes this. Better to fix by deleting code rather than adding.

**Comment by kripken** [comment]
@juj I think that PR is only for `PROXY_TO_PTHREAD` mode, while this is for all pthreads usage? In particular even after that PR there is a read of the variable `noExitRuntime` in `worker.js`, which is incorrect and this fixes.

**Comment by juj** [comment]
Yeah, I see what you mean now. Let me propose a different solution for this once I have a moment, I'd like to not add a new boolean on Module.

File: src/library_pthread.js [hunk 1]
```diff
@@ -1350,6 +1350,11 @@ var LibraryPThread = {
     establishStackSpace(stackTop, stackMax);
   },
 
+  // allow pthreads to check if noExitRuntime from worker.js
+  $checkNoExitRuntime: function() {
```
Comment (juj) [inline src/library_pthread.js]: Maybe just `getNoExitRuntime` as this is a getter? "check" might imply assert or validate?

Comment (kripken) [reply src/library_pthread.js]: Sounds good, renaming.

**Comment by sbc100** [comment]
The deletion of the `incoming` branch triggered the automatic closing of this branch.  My apologies.  This was an unexpected side effect.

Re-opening and re-targeting to master.

**Pull request merged**